### HYPOTHESIS FORMULATION FOR OUR OBJECTIVE ONE.

Load relevant data and csv files :
- The dataset has no missing or duplicate values.
- Numeric columns like Revenue and Budget are in the correct formats 

### Impact of Budget production on Box Office Revenue

For our ttest hypothesis that seek to explain the `financial projections for our investors` 
by:  Budget  comparison ttest:Analyse High-budget ($>$ $17M) vs. low-budget ($\leq$ $17M).

`To evaluate whether high investments on production budgets significantly affect box office performance.A two-sample t-test was conducted comparing revenues from`
- Null Hypothesis (H₀): Budget has no significant impact on box office revenue.
- Alternative Hypothesis (H₁): Higher budgets lead to significantly higher box office revenue


In [1]:
#import necessary libraries
import pandas as pd 
import numpy as np

#### Loading the all combined csv to our `Film_df`

In [2]:
film_df = pd.read_csv("zippedData/all_combined.csv")
film_df.head()

c:\Users\hp\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (19) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


movie_id  ordering  person_id         category  ordering.1          title  \
0  tt0475290        10  nm0005683  cinematographer          24  Hail, Caesar!   
1  tt0475290         1  nm0000982            actor          24  Hail, Caesar!   
2  tt0475290         2  nm0000123            actor          24  Hail, Caesar!   
3  tt0475290         3  nm2403277            actor          24  Hail, Caesar!   
4  tt0475290         4  nm0000146            actor          24  Hail, Caesar!   

  region  is_original_title  primary_title original_title_x  ...  Unnamed: 0  \
0     GB                0.0  Hail, Caesar!    Hail, Caesar!  ...       17497   
1     GB                0.0  Hail, Caesar!    Hail, Caesar!  ...       17497   
2     GB                0.0  Hail, Caesar!    Hail, Caesar!  ...       17497   
3     GB                0.0  Hail, Caesar!    Hail, Caesar!  ...       17497   
4     GB                0.0  Hail, Caesar!    Hail, Caesar!  ...       17497   

        genre_ids    id_y  original_language  original_title_y popularity  \
0  [35, 18, 9648]  270487                 en     Hail, Caesar!     12.312   
1  [35, 18, 9648]  270487                 en     Hail, Caesar!     12.312   
2  [35, 18, 9648]  270487                 en     Hail, Caesar!     12.312   
3  [35, 18, 9648]  270487                 en     Hail, Caesar!     12.312   
4  [35, 18, 9648]  270487                 en     Hail, Caesar!     12.312   

  release_date_y vote_average  vote_count     profits  
0     2016-02-05          5.9        2328  42160680.0  
1     2016-02-05          5.9        2328  42160680.0  
2     2016-02-05          5.9        2328  42160680.0  
3     2016-02-05          5.9        2328  42160680.0  
4     2016-02-05          5.9        2328  42160680.0  

[5 rows x 37 columns]

In [3]:
box_office_df = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
box_office_df

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]


## A.Impact of Budget on Box Office Revenue
`To evaluate whether investments on production budgets significantly affect box office performance`
- Null Hypothesis (H₀): Budget has no significant impact on box office revenue.
- Alternative Hypothesis (H₁): Higher budgets lead to significantly higher box office revenue

In [4]:
#import library
import pandas as pd
#load the csv
budget_df=pd.read_csv('zippedData\cleaned_dataset_movie_budget.csv')
budget_df.head()

id  release_date                                        movie  \
0   1  dec 18, 2009                                       avatar   
1   2  may 20, 2011  pirates of the caribbean: on stranger tides   
2   3   jun 7, 2019                                 dark phoenix   
3   4   may 1, 2015                      avengers: age of ultron   
4   5  dec 15, 2017            star wars ep. viii: the last jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [5]:
budget_df.columns

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

In [6]:
#rename worlwide_gross to revenue
budget_df=budget_df.rename(columns={'worldwide_gross':'revenue'})
print (budget_df.columns)

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'revenue'],
      dtype='object')


In [7]:
#rename production_budget to budget
budget_df=budget_df.rename(columns={'production_budget':'budget'})

Proceed to check for missing values

In [8]:
# Remove non-numeric characters (commas, dollar signs) and convert to numeric
for col in ['budget', 'revenue']:
    budget_df[col] = budget_df[col].astype(str).str.replace(r'[^\d.]', '', regex=True)
    budget_df[col] = pd.to_numeric(budget_df[col], errors='coerce')


In [9]:
#drop rows with missing values
budget_df = budget_df.dropna(subset=['budget', 'revenue'])


Finalise the data cleaning .Assertaining the budget column has no Nan values

In [10]:
# Force all to strings and lowercase
budget_df['budget'] = budget_df['budget'].astype(str).str.lower()

# Replace common bad values with NaN
budget_df['budget'] = budget_df['budget'].replace(['n/a', 'na', '-', 'none', 'unknown', '—'], pd.NA)

# Remove all non-numeric characters (except decimal point)
budget_df['budget'] = budget_df['budget'].str.replace(r'[^\d.]', '', regex=True)

# Convert to numeric
budget_df['budget'] = pd.to_numeric(budget_df['budget'], errors='coerce')




## Lets now Perform the T-Test


let’s `formulate the hypotheses`:

 We will need to obtain High budget and the Low budget threshhold by computation of the median production.This will split by the median 

In [11]:
#COMPUTE THE MEDIAN PRODUCTION FOR THE BUDGET ANALYSIS
# Find the median budget
median_budget = budget_df['budget'].median()
print("Median budget:", median_budget)

# Split into high and low budget groups
low_budget_df = budget_df[budget_df['budget'] < median_budget]
high_budget_df = budget_df[budget_df['budget'] >= median_budget]



Median budget: 17000000.0


 Median Production Budget Analysis
The dataset was cleaned to extract valid numerical values for both budget and revenue. After removing any non-numeric characters and null entries, we computed the median production budget across all films.

Median Budget: $17,000,000

This median value serves as the threshold to divide the dataset into two groups for further statistical analysis:

Low Budget Films: Films with budgets ≤ $17M

High Budget Films: Films with budgets > $17M

This grouping is essential for hypothesis testing to determine whether higher production budgets are associated with significantly higher box office revenues.
We have assertained that the median value of 17,000,000 will be the threshhold of our statistical analysis.
Let us now calculate the T-statistic and P-value

In [12]:
from scipy.stats import ttest_ind

# Extract revenue
low_revenue = low_budget_df['revenue']
high_revenue = high_budget_df['revenue']

# Run Welch’s t-test
t_stat, p_value = ttest_ind(high_revenue, low_revenue, equal_var=False)

# Display results
print("T-Statistic:", t_stat)
print("P-Value:", p_value)

# Interpretation
if p_value <= 0.05:
    print("Reject H₀: High-budget films earn significantly more revenue.")
else:
    print("Fail to reject H₀: No significant revenue difference based on budget.")


T-Statistic: 32.12504153761319
P-Value: 9.18893427925073e-197
Reject H₀: High-budget films earn significantly more revenue.



` The Hypothesis Test: Impact of Budget on Box Office Revenue
To evaluate whether production budgets significantly affect box office performance, a two-sample t-test was conducted comparing revenues from:`

High Budget Films with budgets greater than the 17,000,000 threshold (budget > $17M)

Low Budget Films equal or lesser than the 17,000,000 threshold .(budget ≤ $17M)

 Hypotheses 
Null Hypothesis (H₀): Budget has no significant impact on box office revenue.

Alternative Hypothesis (H₁): Higher budgets lead to significantly higher revenue.

 Test Details
Test Used: Welch’s t-test (accounts for unequal variances)

T-statistic: 32.13

P-value: 9.19 × 10⁻¹⁹⁷

`We thereby Reject the null hypothesis.`

There is overwhelming statistical evidence that higher-budget films earn significantly more revenue than lower-budget films.